### PR TITLE
fix for %USERPROFILE% containing space(s)

### PR DESCRIPTION
**Pull request by valhristov**
fix for ssh-agent not exiting quickly enough and locking files
updated rspec string resources
the DiagnosticDescriptors are now created from resource strings
removed Sqale*Attributes
removing SqaleSubCharacteristic
changing the RuleAttribute constructor
removed TagsAttribute
rewrite RuleDetailBuider

### CODE REVIEW

File: src/Tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj [hunk 1]
```diff
@@ -113,15 +113,16 @@
     </When>
     <Otherwise>
       <ItemGroup>
-        <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework" />
+        <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework">
```
Comment (tvsonar) [inline src/Tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj]: This should probably be reverted.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: .gitignore [hunk 2]
```diff
@@ -63,3 +63,5 @@ packages/
 
 # built Nuget packages
 *.nupkg
+build/cs_strings.restext
+build/vbnet_strings.restext
```
Comment (Evangelink) [inline .gitignore]: Shall we replace by: build/*.restext

File: build/rspec.ps1 [hunk 3]
```diff
@@ -140,46 +125,102 @@ function ToResourceLanguage()
...
+    (
+        [Parameter(Mandatory=$true)]
+        [string]
+        $lang,
+        $rules,
+        $otherLanguageRules
+    )
     $rspecFolder = GetRspecDownloadPath $lang
-    $source_suffix = "_" + $(ToRuleApiLanguage $lang)
+    $source_suffix = "_" + $rule_api_languages.Get_Item($lang)
```
Comment (Evangelink) [inline build/rspec.ps1]: I don't mind using underscore but I think we should be consistent. We are using 'otherLanguageRules' and 'source_suffix'

File: rspec/cs/S1109_c#.html [hunk 4]
```diff
@@ -2,12 +2,14 @@
 beginning of a line.</p>
 <h2>Noncompliant Code Example</h2>
 <pre>
-if(condition) {
+if(condition)
```
Comment (Evangelink) [inline rspec/cs/S1109_c#.html]: Add space after if

File: rspec/cs/S1109_c#.html [hunk 5]
```diff
@@ -2,12 +2,14 @@
...
+{
   doSomething();}
 </pre>
 <h2>Compliant Solution</h2>
 <pre>
-if(condition) {
+if(condition)
```
Comment (Evangelink) [inline rspec/cs/S1109_c#.html]: Add space after if

File: rspec/cs/S1185_c#.html [hunk 6]
```diff
@@ -15,4 +15,28 @@
   //do something else
 }
 </pre>
+<h2>Exceptions</h2>
+<p>If there is an attribute in any level of the overriding chain, then the overridden member is ignored.</p>
+<pre>
+public class Base
+{
+  [Required]
+  public virtual string Name { get; set }
```
Comment (Evangelink) [inline rspec/cs/S1185_c#.html]: Missing ';' after the set

File: rspec/cs/S122_c#.html [hunk 7]
```diff
@@ -1,12 +1,13 @@
 <p>For better readability, do not put more than one statement on a single line.</p>
 <h2>Noncompliant Code Example</h2>
 <pre>
-if(someCondition) doSomething();
+if(someCondition) DoSomething();
```
Comment (Evangelink) [inline rspec/cs/S122_c#.html]: Add space after if

File: rspec/cs/S122_c#.html [hunk 8]
```diff
@@ -1,12 +1,13 @@
...
 </pre>
 <h2>Compliant Solution</h2>
 <pre>
-if(someCondition) {
-  doSomething();
+if(someCondition)
```
Comment (Evangelink) [inline rspec/cs/S122_c#.html]: Add space after if

File: rspec/cs/S2758_c#.html [hunk 9]
```diff
@@ -2,14 +2,16 @@
...
+  return person.GetAge() &gt; 18 ? true : true; // Noncompliant; is this what was intended?
 }
 </pre>
 <h2>Compliant Solution</h2>
 <pre>
-public boolean canVote(Person person) {
-  return person.getAge &gt; 18 ? true : false;
+public bool CanVote(Person person)
+{
+  return person.GetAge() &gt; 18 ? true : false;
```
Comment (Evangelink) [inline rspec/cs/S2758_c#.html]: Is it on purpose to give a compliant solution leading to another issue? `return person.GetAge() > 18` should be enough.

Comment (tvsonar) [inline rspec/cs/S2758_c#.html]: I'm fixing this in RSPEC.

File: rspec/cs/S1185_c#.html (hunk 6 above)
Comment (tvsonar) [inline rspec/cs/S1185_c#.html]: I'm fixing this in RSPEC

**Comment by tvsonar** [comment]
@Evangelink We tend to use 2 spaces in RSPEC for indenting. So we should stick to that in code samples.

**Pull request merged**